modules/javafx.swing/src/main/java/com/sun/javafx/embed/swing/SwingEvents.java

```diff
@@ -81,16 +81,18 @@ public static int mouseButtonToEmbedMouseButton(int button, int extModifiers) {
                 break;
         }
         // Fix for RT-15457: we should report mouse buttons for mouse drags
-        if ((extModifiers & MouseEvent.BUTTON1_DOWN_MASK) != 0) {
-            abstractButton = AbstractEvents.MOUSEEVENT_PRIMARY_BUTTON;
-        } else if ((extModifiers & MouseEvent.BUTTON2_DOWN_MASK) != 0) {
-            abstractButton = AbstractEvents.MOUSEEVENT_MIDDLE_BUTTON;
-        } else if ((extModifiers & MouseEvent.BUTTON3_DOWN_MASK) != 0) {
-            abstractButton = AbstractEvents.MOUSEEVENT_SECONDARY_BUTTON;
-        } else if ((extModifiers & MouseEvent.getMaskForButton(4)) != 0) {
-            abstractButton = AbstractEvents.MOUSEEVENT_BACK_BUTTON;
-        } else if ((extModifiers & MouseEvent.getMaskForButton(5)) != 0) {
-            abstractButton = AbstractEvents.MOUSEEVENT_FORWARD_BUTTON;
+        if (abstractButton == AbstractEvents.MOUSEEVENT_NONE_BUTTON) {
```

Comment:
minor: the placement of the old comment suggests that it applies to like 84 (this change).  Perhaps the comment should be moved down to the line 85.

Reply:
ok..done..